pkg/runner/job_runner.go

```diff
@@ -256,10 +271,54 @@ func (jr *JobRunner) Run(j *job.Job) ([][]*job.Report, []*job.Report, error) {
 	return allRunsReports, finalReports, nil
 }
 
+// GetCurrentRun returns the run which is currently being executed
+func (jr *JobRunner) GetCurrentRun(jobID types.JobID) (*types.RunID, error) {
+	runEvents, err := jr.eventManager.Fetch(
+		[]frameworkevent.QueryField{
+			frameworkevent.QueryJobID(jobID),
+			frameworkevent.QueryEventName(EventRunStarted),
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch last run id for job %d: %v", jobID, err)
+	}
+
+	lastEvent := runEvents[len(runEvents)-1]
+	payload := runStartedPayload{}
+	if err := json.Unmarshal([]byte(*lastEvent.Payload), &payload); err != nil {
+		return nil, fmt.Errorf("could not fetch last run id for job %d: %v", jobID, err)
+	}
+	return &payload.RunID, nil
+
+}
+
+func (jr *JobRunner) emitEvent(jobID types.JobID, eventName event.Name, payload interface{}) error {
+	payloadJSON, err := json.Marshal(payload)
+	if err != nil {
+		jobLog.Warningf("could not encode payload for event %s: %v", eventName, err)
+		return err
+	}
+
+	rawPayload := json.RawMessage(payloadJSON)
+	ev := frameworkevent.Event{JobID: jobID, EventName: eventName, Payload: &rawPayload, EmitTime: time.Now()}
+	if err := jr.eventManager.Emit(ev); err != nil {
+		jobLog.Warningf("could not emit event %s: %v", eventName, err)
+		return err
+	}
+	if err := jr.eventManager.Emit(ev); err != nil {
```

Comment:
**Important**

Why do you `Emit` it twice?

Reply:
Bug.